src/TestApplication.cpp

```diff
@@ -2,10 +2,10 @@
 
 TestApplication::TestApplication(int& argc, char** argv)
 : QApplication(argc, argv)
-, serialPort_(new QSerialPort())
 , vehicleData_(new VehicleData())
-, telemetryReporting_(new TelemetryReporting(*serialPort_, *vehicleData_))
-, window_(new Window(*serialPort_, *telemetryReporting_))
+, view_(new View())
+, communicationService_(new CommunicationService(*view_))
+, telemetryReporting_(new TelemetryReporting(communicationService_->getPeripheral(), *vehicleData_, *view_))
```

Comment:
Makes sense, let's have a ConnectionService, which will use serial connection service and (udp connection service). Depending on what button is pressed, ConnectionService will use either more specialized connectionservices.